ArduSub/AP_Arming_Sub.cpp

```diff
@@ -53,6 +53,37 @@ bool AP_Arming_Sub::arm(AP_Arming::Method method, bool do_arming_checks)
         return false;
     }
 
+    // if there's no shift assigned, it must not be a shift funtion
+    bool has_shift_btn = false;
+    bool has_disarm_btn = false;
+    // make sure the craft has a disarm button assigned before it is armed
+    for (int i = 0; i < 32; i++) {
+        int btn;
+        // check all the standard btn functions and then check all the shift functions
+        if (i < 16) {
+            btn = sub.get_button(i)->function(false);
+        } else {
+            // dont bother checking the rest if there's no shift assigned
+            if (!has_shift_btn) {
+                break;
+            }
+            btn = sub.get_button(i - 16)->function(true);
+        }
+
+        if (btn == JSButton::k_arm_toggle || btn == JSButton::k_disarm) {
+            has_disarm_btn = true;
+            break;
+        } else if (btn == JSButton::k_shift) {
+            has_shift_btn = true;
+        }
+    }
+
+    // don't allow arming unless there is a disarm button configured
+    if (!has_disarm_btn) {
+        gcs().send_text(MAV_SEVERITY_NOTICE,"Must assign a disarm or arm_toggle button");
```

Comment:
`NOTICE` may not be a high-enough level here.

By the time you are calling `check_failed` that isn't your concern, however.